### PR TITLE
mainEvents: drop deprecated emit/on

### PR DESCRIPTION
**Pull request by mook-as**
Drop the generic versions of `emit()` and `on()`, which are confusing because it's not always clear why they are deprecated.  Instead, implement the `MainEvents` interface by forwarding to the methods from `EventEmitter` manually.  This should reduce confusion about why a call isn't working (since TypeScript can better guess which one is expected and emit better errors about any arguments that are incorrect).

### CODE REVIEW

**Comment by ericpromislow** [comment]
JetBrains, which doesn't have the most sophisticated typescript linter around, replies with this an unreadable blob of HTML. But the gist is that one side has the type 

`RecursivePartial<settings.Settings>` and the other side has the type
`RecursivePartial<<settings.Settings>>`

Changing the signature of `writeSettings` to

`function writeSettings(arg: RecursivePartial<RecursiveReadonly<settings.Settings>>) {`

"fixes" this, and makes sense, as `arg` is not modified.

**Comment by ericpromislow** [comment]
No need to do the same with `mainEvents.once`?  It isn't defined in, so I guess it's defined in terms of `mainEvents.on` ?

**Review by ericpromislow — CHANGES_REQUESTED** [review]
I'm going by my IDE more than my understanding, but think that the `writeSettings` signature should be modified.

**Review by rak-phillip — DISMISSED** [review]
These changes look good to me and I'm glad that we are able to move away from the deprecation warnings. I agree that those were always confusing. 

I'm not experiencing any linter or in-editor warnings (vscode/webstorm) before or after rebasing this change onto main.

**Review by rak-phillip — APPROVED** [review]
LGTM - I think that `once` is a good inclusion. No linter or in-editor (vscode, webstorm) warnings related to this change.

**Review by rak-phillip — CHANGES_REQUESTED** [review]
e2e test errors uncovered that one of the changes included in this PR breaks the preferences window:

```
Error in fetch(): TypeError: Cannot read properties of undefined (reading 'port')
    at _callee$ (preferences.ts?dbd3:122:1)
    at tryCatch (runtime.js?96cf:63:1)
    at Generator.invoke [as _invoke] (runtime.js?96cf:294:1)
    at Generator.eval [as next] (runtime.js?96cf:119:1)
    at asyncGeneratorStep (asyncToGenerator.js?1da1:3:1)
    at _next (asyncToGenerator.js?1da1:25:1)
    at eval (asyncToGenerator.js?1da1:32:1)
    at new Promise (<anonymous>)
    at eval (asyncToGenerator.js?1da1:21:1)
    at Store.fetchPreferences (preferences.ts?dbd3:141:1)
```

File: background.ts
Comment (rak-phillip) [inline background.ts:527]: Something about this change causes `fetchCredentials` to return `undefined`

https://github.com/rancher-sandbox/rancher-desktop/blob/50188b15e989abb43ee822e73647a0975e69d71b/pkg/rancher-desktop/store/credentials.ts#L31-L36

**Comment by mook-as** [comment]
Fixed the failure (yay E2E tests!) by switching getting API credentials to just using `invoke`/`handle`. Things get confused when the same event is used for both `on` and `handle`.

**Review by rak-phillip — DISMISSED** [review]
LGTM - we need to rebase before merging.

**Comment by mook-as** [comment]
Somehow rebasing lead to more stuff around the credentials vuex store being flagged; sorry about that.

**Review by rak-phillip — APPROVED** [review]
Latest round of updates looks good.

**Review by ericpromislow — APPROVED** [review]
Looks good

**Pull request merged**
